### PR TITLE
Add blog post describing PSP deprecation and next steps.

### PR DESCRIPTION
**Pull request by tabbysable**
Add blog post describing PSP deprecation and next steps, to come out before or with the 1.21 release [[preview](https://deploy-preview-27369--kubernetes-io-master-staging.netlify.app/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/)]

### CODE REVIEW

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Whenever a Kubernetes feature is set to go away, our [deprecation policy](docs/reference/using-api/deprecation-policy/) is our guide. First the feature is marked as deprecated, then after enough time has passed, it can finally be removed.
+
+Kubernetes 1.21 starts the deprecation process for PodSecurityPolicy. As with all feature deprecations, PodSecurityPolicy will continue to be fully functional for several more releases. The current plan is to remove PSP from Kubernetes in the 1.25 release.
+
+Until then, PSP is still PSP. There will be at least a year during which the newest Kubernetes releases will still support PSP, and nearly two years until PSP will pass fully out of all supported Kubernetes versions.
+
+## What is PodSecurityPolicy?
+
+[PodSecurityPolicy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) is a built-in [admission controller](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) that allows a cluster administrator to control security-sensitive aspects of the Pod specification.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:24]: nit
```suggestion
[PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/) is a built-in [admission controller](o/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) that allows a cluster administrator to control security-sensitive aspects of the Pod specification. 
```
We use site-relative hyperlinks.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Until then, PSP is still PSP. There will be at least a year during which the newest Kubernetes releases will still support PSP, and nearly two years until PSP will pass fully out of all supported Kubernetes versions.
+
+## What is PodSecurityPolicy?
+
+[PodSecurityPolicy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) is a built-in [admission controller](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) that allows a cluster administrator to control security-sensitive aspects of the Pod specification. 
+
+First, one or more PodSecurityPolicy resources are created in a cluster to define the requirements Pods must meet. Then, RBAC rules are created to control which PodSecurityPolicy applies to a given pod. If a pod meets the requirements of its PSP, it will be admitted to the cluster as usual. In some cases, PSP can also modify Pod fields, effectively creating new defaults for those fields. If a Pod does not meet the PSP requirements, it is rejected, and cannot run. 
+
+One more important thing to know about PodSecurityPolicy: it’s not the same as PodSecurityContext.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:28]:
```suggestion
One more important thing to know about PodSecurityPolicy: it’s not the same as [PodSecurityContext](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context).
```
?

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+The deprecation of PSP does not affect PodSecurityContext in any way.
+
+## Why did we need PodSecurityPolicy?
+
+In Kubernetes, we define resources such as Deployments, StatefulSets, and Services that represent the building blocks of software applications. The various controllers inside a Kubernetes cluster react to these resources, creating further Kubernetes resources or configuring some software or hardware to accomplish our goals.
+
+In most Kubernetes clusters, RBAC (Role-Based Access Control) rules control access to these resources. Get, Create, Edit, and Delete are the sorts of operations that RBAC cares about, but RBAC does not consider what settings are being put into the resources it controls. For example, a Pod can be almost anything from a simple webserver to a privileged command prompt offering full access to the underlying server node and all the data. It’s all the same to RBAC: a Pod is a Pod is a Pod.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:38]: I'd write:
```suggestion
In most Kubernetes clusters, RBAC (Role-Based Access Control) rules control access to these resources. `list`, `get`, `create`, `edit`, and `delete` are the sorts of API operations that RBAC cares about, but _RBAC does not consider what settings are being put into the resources it controls_. For example, a Pod can be almost anything from a simple webserver to a privileged command prompt offering full access to the underlying server node and all the data. It’s all the same to RBAC: a Pod is a Pod is a Pod.
```
I added `list` to give a strong hint that these aren't HTTP verb names.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Today, you’re not limited only to deploying PSP or writing your own custom admission controller. Several external admission controllers are available that incorporate lessons learned from PSP to provide a better user experience. [OPA/Gatekeeper](https://github.com/open-policy-agent/gatekeeper/), [Kyverno](https://github.com/kyverno/kyverno/), and [K-Rail](https://github.com/cruise-automation/k-rail) are all well-known, and each has its fans.
+
+Although there are other good options available now, we believe there is still value in having a built-in admission controller available as a choice for users. With this in mind, we turn toward building what’s next, inspired by the lessons learned from PSP.
+
+## What’s next?
+
+Kubernetes SIG Security, SIG Auth, and a diverse collection of other community members have been working together for months to ensure that what’s coming next is going to be awesome. We have developed a Kubernetes Enhancement Proposal ([KEP 2579](https://github.com/kubernetes/enhancements/issues/2579)) and a prototype for a new feature, currently being called PSP Replacement Policy. We are targeting an Alpha release in Kubernetes 1.22.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:56]: Does it go without saying that _PSP Replacement Policy_ is a working title?

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+If you’re making extensive use of the flexibility of PSP with numerous PSPs and complex binding rules, you will likely find the simplicity of PSP Replacement Policy too limiting. Use the next year to evaluate the other admission controller choices in the ecosystem. There are resources available to ease this transition, such as the [Gatekeeper Policy Library](https://github.com/open-policy-agent/gatekeeper-library).
+
+If your use of PSP is relatively simple, with a few policies and straightforward binding to service accounts in each namespace, you will likely find PSP Replacement Policy to be a good match for your needs. Evaluate your PSPs compared to the [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) to get a feel for where you’ll be able to use the Restricted, Baseline, and Privileged policies. Please follow along with or contribute to the KEP and subsequent development, and try out the Alpha release of PSP Replacement Policy when it becomes available.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:68]: I'd write:
```suggestion
If your use of PSP is relatively simple, with a few policies and straightforward binding to service accounts in each namespace, you will likely find PSP Replacement Policy to be a good match for your needs. Evaluate your PSPs compared to the Kubernetes [Pod Security Standards](/docs/concepts/security/pod-security-standards/) to get a feel for where you’ll be able to use the Restricted, Baseline, and Privileged policies. Please follow along with or contribute to the KEP and subsequent development, and try out the Alpha release of PSP Replacement Policy when it becomes available.
```

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+If you’re just beginning your PSP journey, you will save time and effort by keeping it simple. You can approximate the functionality of PSP Replacement Policy today by using the Pod Security Standards’ PSPs. If you set the cluster default by binding a Baseline or Restricted policy to the system:authenticated group, and then make a more-permissive policy available as needed in certain Namespaces [using ServiceAccount bindings](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#run-another-pod), you will avoid many of the PSP pitfalls and have an easy migration to PSP Replacement Policy. If your needs are much more complex than this, your effort is probably better spent adopting one of the more fully-featured external admission controllers mentioned above.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:70]:
```suggestion
If you’re just beginning your PSP journey, you will save time and effort by keeping it simple. You can approximate the functionality of PSP Replacement Policy today by using the Pod Security Standards’ PSPs. If you set the cluster default by binding a Baseline or Restricted policy to the system:authenticated group, and then make a more-permissive policy available as needed in certain Namespaces [using ServiceAccount bindings](/docs/concepts/policy/pod-security-policy/#run-another-pod), you will avoid many of the PSP pitfalls and have an easy migration to PSP Replacement Policy. If your needs are much more complex than this, your effort is probably better spent adopting one of the more fully-featured external admission controllers mentioned above.
```

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+---
+layout: blog
+title: "PodSecurityPolicy: Past, Present, and Future"
+date: 2021-04-08
+slug: PodSecurityPolicy-Past-Present-and-Future
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:5]:
```suggestion
slug: podsecuritypolicy-past-present-and-future
```

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+**Author:** Tabitha Sable (Kubernetes SIG Security)
+
+PodSecurityPolicy (PSP) is being deprecated in Kubernetes 1.21. This starts the countdown to its removal, but doesn’t change anything else. PodSecurityPolicy will continue to be fully functional for several more releases before being removed completely. In the meantime, we are developing a replacement for PSP that covers key use cases more easily and sustainably.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:10]: Should we mention “released today” in connection with the Kubernetes 1.21 reference? The current article date is for the release day.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+What are Pod Security Policies? Why did we need them? Why are they going away, and what’s next? How does this affect you? These key questions come to mind as we prepare to say goodbye to PSP, so let’s walk through them together. We’ll start with an overview of how features get removed from Kubernetes.
+
+## What does deprecation mean in Kubernetes?
+
+Whenever a Kubernetes feature is set to go away, our [deprecation policy](docs/reference/using-api/deprecation-policy/) is our guide. First the feature is marked as deprecated, then after enough time has passed, it can finally be removed.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:16]: :warning: broken hyperlink
```suggestion
Whenever a Kubernetes feature is set to go away, our [deprecation policy](/docs/reference/using-api/deprecation-policy/) is our guide. First the feature is marked as deprecated, then after enough time has passed, it can finally be removed.
```

**Review by sftim — COMMENTED** [review]
@tabbysable thanks for this!

Here's some feedback - the main thing is to use site-relative links, and there's a few inline things too that are more like suggestions.

**Comment by sftim** [comment]
I edited the PR description to include a link to https://deploy-preview-27369--kubernetes-io-master-staging.netlify.app/blog/2021/04/08/podsecuritypolicy-past-present-and-future/

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
+---
+layout: blog
+title: "PodSecurityPolicy: Past, Present, and Future"
+date: 2021-04-08
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:4]: If you're not sure about the date but that's the only thing to fix, you can add `draft: true` to the front matter and send in another PR to remove that once the date is confirmed.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
+---
+layout: blog
+title: "PodSecurityPolicy: Past, Present, and Future"
```
Comment (mbbroberg) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:3]: This is a wonderful read! Tim covered the review, and I had just one minor addition from years of SEO work. 

This article will show up more often in Google/Bing/DDG searches and help more people if it's title and slug align to a searcher's intent. My assumption there is people will search `PSP deprecation` or PodSecurityPolicy deprecation`, so it'd be great to get those terms in both places. If you want to keep a similar format: `the How, What, and When of PSP Deprecation` could be fun. Or something playful like `Don't Panic: Updates on PodSecurityPolicy Deprecation`. Either way, the keyword in the title (and slug) helps a lot with ranking.

FWIW as well, I found this article is in the top position on search and it's got some great additional context. Adding here for others to see: https://www.antitree.com/2020/11/pod-security-policies-are-being-deprecated-in-kubernetes/

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 4 above)
Comment (tabbysable) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:56]: I think that's pretty clearly implied by the "currently" but could also see changing to "new feature with a temporary working title of PSP Replacement Policy". Thoughts?

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 7 above)
Comment (tabbysable) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:5]: holding off on addressing this pending potential retitling

**Comment by tabbysable** [comment]
@divya-mohan0209 What do you think about a publication date for this?

I like the idea of getting it out a couple of days before the release so that our first official communication about it won't be the deprecation release note, and so the release note can refer to this post right away.

**Comment by divya-mohan0209** [comment]
@tabbysable I think getting it out a couple of days before the release would be ideal given that we could reference it in the Release blog too. Probably 6th April would be a good publish date.

@palnabarun WDYT?

**Comment by tabbysable** [comment]
I like the idea of the 6th -- still gives time for people to review, but gets it out before the release. I'll update the branch with that later today unless there's strong opposition.

**Comment by sftim** [comment]
@tabbysable, @liggitt, @tallclair, @IanColdwater  - what do you think about merging #26629 the same day that the blog article publishes?

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+date: 2021-04-08
+slug: PodSecurityPolicy-Past-Present-and-Future
+---
+
+**Author:** Tabitha Sable (Kubernetes SIG Security)
+
+PodSecurityPolicy (PSP) is being deprecated in Kubernetes 1.21. This starts the countdown to its removal, but doesn’t change anything else. PodSecurityPolicy will continue to be fully functional for several more releases before being removed completely. In the meantime, we are developing a replacement for PSP that covers key use cases more easily and sustainably.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:10]: PodSecurityPolicy is _also_ the `kind` of an object in the API you use to configure that admission controller's behavior. Some people will only have seen the API end of things, not the kube-apiserver configuration that makes it work.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+Until then, PSP is still PSP. There will be at least a year during which the newest Kubernetes releases will still support PSP, and nearly two years until PSP will pass fully out of all supported Kubernetes versions.
+
+## What is PodSecurityPolicy?
+
+[PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/) is a built-in [admission controller](/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) that allows a cluster administrator to control security-sensitive aspects of the Pod specification. 
+
+First, one or more PodSecurityPolicy resources are created in a cluster to define the requirements Pods must meet. Then, RBAC rules are created to control which PodSecurityPolicy applies to a given pod. If a pod meets the requirements of its PSP, it will be admitted to the cluster as usual. In some cases, PSP can also modify Pod fields, effectively creating new defaults for those fields. If a Pod does not meet the PSP requirements, it is rejected, and cannot run. 
+
+One more important thing to know about PodSecurityPolicy: it’s not the same as [PodSecurityContext](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context).
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:28]:
```suggestion
One more important thing to know about PodSecurityPolicy: it’s not the same as the [`securityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context) for a Pod or container.
```

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+The way PSPs are applied to Pods has proven confusing to nearly everyone that has attempted to use them. It is easy to accidentally grant broader permissions than intended, and difficult to inspect which PSP(s) apply in a given situation. The “changing Pod defaults” feature can be handy, but is only supported for certain Pod settings and it’s not obvious when they will or will not apply to your Pod. Without a “dry run” or audit mode, it’s impractical to retrofit PSP to existing clusters safely, and it’s impossible for PSP to ever be enabled by default.
+
+For more information about these and other PSP difficulties, check out SIG Auth’s KubeCon NA 2019 Maintainer Track session video: {{< youtube "SFtHRmPuhEw?start=953" youtube-quote-sm >}}
+
+Today, you’re not limited only to deploying PSP or writing your own custom admission controller. Several external admission controllers are available that incorporate lessons learned from PSP to provide a better user experience. [OPA/Gatekeeper](https://github.com/open-policy-agent/gatekeeper/), [Kyverno](https://github.com/kyverno/kyverno/), and [K-Rail](https://github.com/cruise-automation/k-rail) are all well-known, and each has its fans.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:50]: aside: CNCF policy on listing products that include 3rd party options is to write them in alphabetical order.

As this is a _blog post_ I think it's important to have regard for that policy but if we want to express a preference, we (specifically @tabbysable as article author) can absolutely do that.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 4 above)
Comment (sftim) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:56]: (whichever you prefer!)

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+What are Pod Security Policies? Why did we need them? Why are they going away, and what’s next? How does this affect you? These key questions come to mind as we prepare to say goodbye to PSP, so let’s walk through them together. We’ll start with an overview of how features get removed from Kubernetes.
+
+## What does deprecation mean in Kubernetes?
+
+Whenever a Kubernetes feature is set to go away, our [deprecation policy](/docs/reference/using-api/deprecation-policy/) is our guide. First the feature is marked as deprecated, then after enough time has passed, it can finally be removed.
+
+Kubernetes 1.21 starts the deprecation process for PodSecurityPolicy. As with all feature deprecations, PodSecurityPolicy will continue to be fully functional for several more releases. The current plan is to remove PSP from Kubernetes in the 1.25 release.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:18]: If this article goes out before the v1.21 release, let's say something along the lines of “due for release later this week”.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+The deprecation of PSP does not affect PodSecurityContext in any way.
+
+## Why did we need PodSecurityPolicy?
+
+In Kubernetes, we define resources such as Deployments, StatefulSets, and Services that represent the building blocks of software applications. The various controllers inside a Kubernetes cluster react to these resources, creating further Kubernetes resources or configuring some software or hardware to accomplish our goals.
+
+In most Kubernetes clusters, RBAC (Role-Based Access Control) rules control access to these resources. `list`, `get`, `create`, `edit`, and `delete` are the sorts of API operations that RBAC cares about, but _RBAC does not consider what settings are being put into the resources it controls_. For example, a Pod can be almost anything from a simple webserver to a privileged command prompt offering full access to the underlying server node and all the data. It’s all the same to RBAC: a Pod is a Pod is a Pod.
```
Comment (sftim) [inline content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:38]: _optionally_: link to https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 7 above)
Comment (sftim) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:5]: If changing the title, we might need to revise the Kubernetes v1.21 release blog to match.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 13 above)
Comment (tabbysable) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md:28]: Given that the YAML is `securityContext` but the API name is `PodSecurityContext`, and it's the API name that is confusingly similar looking to `PodSecurityPolicy`, I think this edit obscures the meaning of the warning/note here.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 7 above)
Comment (tabbysable) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md]: I've updated the title, please update the release blog to match.

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 14 above)
Comment (tabbysable) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md]: I'd actually like not to express a preference, thanks for calling out that policy.

**Comment by tabbysable** [comment]
I've adopted most of the fine-tuning suggestions @sftim had made.
/unhold

**Comment by sftim** [comment]
(more LGTMs welcome - in SIG Docs, we typically _don't_ approve changes to the live website without two separate reviewers checking the change is OK).

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 7 above)
Comment (divya-mohan0209) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md]: Ack, done :)

**Comment by IanColdwater** [comment]
/lgtm 🥳

Thank you for doing the good work, everybody!

File: content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md (hunk 11 above)
Comment (IanColdwater) [reply content/en/blog/_posts/2021-04-08-PodSecurityPolicy-Past-Present-and-Future.md]: Thank you for the reminder about that blog post! It has some misinformation in it, so I sent a note to Mark to correct it. 😊

File: content/en/blog/_posts/2021-04-06-PodSecurityPolicy-Past-Present-and-Future.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+First, one or more PodSecurityPolicy resources are created in a cluster to define the requirements Pods must meet. Then, RBAC rules are created to control which PodSecurityPolicy applies to a given pod. If a pod meets the requirements of its PSP, it will be admitted to the cluster as usual. In some cases, PSP can also modify Pod fields, effectively creating new defaults for those fields. If a Pod does not meet the PSP requirements, it is rejected, and cannot run. 
+
+One more important thing to know about PodSecurityPolicy: it’s not the same as [PodSecurityContext](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context).
+
+A part of the Pod specification, PodSecurityContext is the collection of fields that specify many of the security-relevant settings for a Pod. The security context dictates to the kubelet and container runtime how the Pod should actually be run. In contrast, the PodSecurityPolicy only constrains (or defaults) the values that may be set on the security context.
```
Comment (tallclair) [inline content/en/blog/_posts/2021-04-06-PodSecurityPolicy-Past-Present-and-Future.md:30]:
```suggestion
A part of the Pod specification, PodSecurityContext (and its per-container counterpart `SecurityContext`) is the collection of fields that specify many of the security-relevant settings for a Pod. The security context dictates to the kubelet and container runtime how the Pod should actually be run. In contrast, the PodSecurityPolicy only constrains (or defaults) the values that may be set on the security context.
```

File: content/en/blog/_posts/2021-04-06-PodSecurityPolicy-Past-Present-and-Future.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+If you’re just beginning your PSP journey, you will save time and effort by keeping it simple. You can approximate the functionality of PSP Replacement Policy today by using the Pod Security Standards’ PSPs. If you set the cluster default by binding a Baseline or Restricted policy to the system:authenticated group, and then make a more-permissive policy available as needed in certain Namespaces [using ServiceAccount bindings](/docs/concepts/policy/pod-security-policy/#run-another-pod), you will avoid many of the PSP pitfalls and have an easy migration to PSP Replacement Policy. If your needs are much more complex than this, your effort is probably better spent adopting one of the more fully-featured external admission controllers mentioned above.
```
Comment (tallclair) [inline content/en/blog/_posts/2021-04-06-PodSecurityPolicy-Past-Present-and-Future.md:70]: nit: In practice these should grant equivalent permission, but I prefer the slightly more specific version.

```suggestion
If you’re just beginning your PSP journey, you will save time and effort by keeping it simple. You can approximate the functionality of PSP Replacement Policy today by using the Pod Security Standards’ PSPs. If you set the cluster default by binding a Baseline or Restricted policy to the `system:serviceaccounts` group, and then make a more-permissive policy available as needed in certain Namespaces [using ServiceAccount bindings](/docs/concepts/policy/pod-security-policy/#run-another-pod), you will avoid many of the PSP pitfalls and have an easy migration to PSP Replacement Policy. If your needs are much more complex than this, your effort is probably better spent adopting one of the more fully-featured external admission controllers mentioned above.
```

**Review by tallclair — COMMENTED** [review]
/lgtm

With optional suggestions.

**Comment by pmmalinov01** [comment]
/lgtm
Will the publish date on the blog post change if the release is delayed?

**Comment by divya-mohan0209** [comment]
> /lgtm
> Will the publish date on the blog post change if the release is delayed?

No @pmmalinov01 . This was supposed to go ahead of the Release & provided we have all the approvals in place, the post will be published on 6th April, 2021.

**Comment by sftim** [comment]
Seems better to publish than wait.
/approve
